encrypter.cpp

```diff
@@ -0,0 +1,202 @@
+    put_error(error_message);
+  }
+
+  char inp[(long int)(size)];
+  file.seekg(0, std::ios::beg);
+  file.read(inp, size);
+  file.close();
+  std::vector<char> input(inp, inp + size);
+
+  if (size == 0)
+    put_error("Given file is empty\n");
+
+  return input;
+}
+
+//=====write_file=====//
+void write_file(const std::string& filename, std::vector<char> output, const std::string error_message, std::streampos& size)
+{
+  std::ofstream file (filename, std::ios::out|std::ios::binary|std::ios::trunc);
+  if (!file.is_open())
+  {
+    put_error(error_message);
+  }
+  
+  file.write(&output[0], size);
+}
+
+//=====make_key=====//
+void make_key(const std::string& filename)
+{
+  std::streampos size;
+  std::vector<char> input = read_file(filename, "Couldn't access file-base for a key.\n", size);
+  shuffle(input);
+  std::string keyname = replace_extension(filename, "key");
+  write_file(keyname, input, "Couldn't open file to save the result.\n", size);
+}
+
+//=====read_key_position=====//
+std::streampos read_key_position(const std::string keyname)
+{
+  std::streampos start = 0;
+  std::ifstream meta (keyname + ".meta");
+  if (meta.is_open())
+  {
+    long int tmp = 0;
+    meta >> tmp;
+    start = tmp;
+  }
+
+  return start;
+}
+
+//=====record_meta=====//
+void record_meta(const long int key_starts, const long int size_of_file, const std::string keyname, std::ofstream& out)
+{
+  std::ofstream meta (keyname + ".meta", std::ios::out|std::ios::trunc);
+  meta << key_starts + size_of_file;
+
+  std::string tmp = std::to_string(key_starts) + "\n";
+  out.write(tmp.c_str(), tmp.length());
+}
+
+//=====read_meta=====//
+std::streampos read_meta(std::ifstream& file)
+{
+  std::string tmp;
+  std::getline(file, tmp);
+  std::streampos key_starts = std::stol (tmp, nullptr);
+
+  return key_starts; 
+}
+
+//=====crypting=====//
+void crypting(const std::string& filename, const std::string& keyname, bool encrypt = true)
+{
+  std::streampos key_starts;
+  std::ifstream file (filename, std::ios::in|std::ios::binary|std::ios::ate);
+  std::ifstream key (keyname, std::ios::in|std::ios::binary|std::ios::ate);
+  if (!file.is_open() || !key.is_open())
+  {
+    put_error("Couldn't open input and/or key files.\n");
+  }
+  std::ofstream out;
+  if (encrypt)
+  {
+    out.open(filename + ".enc", std::ios::out|std::ios::binary);
+    key_starts = read_key_position(keyname);
+    if ((long int)(key.tellg()) - key_starts + 1 < file.tellg())
+      put_error("The key needs to be changed. There is not enough data to encrypt file safely.\n");
+    record_meta(key_starts, file.tellg(), keyname, out);
+    file.seekg(0, std::ios::beg);
+  }
+  else
+  {
+    file.seekg(0, std::ios::beg);
+    key_starts = read_meta(file);
+    out.open(filename.substr(0, filename.length() - 4), std::ios::out|std::ios::binary);
+  }
+  key.seekg(key_starts, std::ios::beg);
+  if (!out.is_open())
+  {
+    put_error("Couldn't open output file.\n");
+  }
+
+  char orig;
+  
```

Comment:
remove empty line before while. "char orig" is virtually a part of while loop here.